### PR TITLE
[Customize Your Store] Switch Theme to TT3

### LINKED ISSUE #40462: [Customize Your Store] Switch Theme to TT23
### User Story

> As a new merchant, I want to be presented with simple flows to customize my store, so that I can start selling sooner.

### Acceptance Criteria

```
Feature: Customize Your Store AI Wizard Loading Screen

  Scenario: AI Wizard loading screen is entered
    Given the AI Wizard loading screen is active
    Then the theme should be switched to Twenty Twenty-Three
```

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/40462.

Switch Theme to TT3 during CYS AI Wizard loading screen and remove swtich_theme "action" so that task won't be marked complete.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

This feature is part of the larger customize your store development and behind a feature flag, hence QA review can be deferred until the call for testing before release.

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

1. Create a new WooCommerce installation with this version.
2. Install  WooCommerce Block [Nightly](https://github.com/woocommerce/woocommerce-blocks/releases/tag/nightly)
3. Make sure to enable `customize-store` feature flag
4. Switch the theme to a block theme other than TT3
5. Delete `woocommerce_admin_customize_store_completed` option 
6. Go to `WooCommerce -> Home` and click `Customize Store` task
7. Click `Design with AI` button
8. Fill in the business description and click on 'continue'
9. Proceed with the rest of the AI wizard
10. After reaching the site assembler, go to `/wp-admin/themes.php`
11. Confirm theme is switched to `Twenty Twenty-Three`
12. Go to `WooCommerce -> Home`
13. Observe that `Customize Store` task is still active.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php [hunk 1]
```diff
@@ -18,7 +18,8 @@ public function __construct( $task_list ) {
 		parent::__construct( $task_list );
 
 		add_action( 'admin_enqueue_scripts', array( $this, 'possibly_add_site_editor_scripts' ) );
-		add_action( 'after_switch_theme', array( $this, 'mark_task_as_complete' ) );
+		// Use "switch_theme" instead of "after_switch_theme" because the latter is fired after the next WP load and we need to remove action when switching theme to TT3 via onboarding theme API. See Automattic\WooCommerce\Admin\API\OnboardingThemes::activate_theme().
+		add_action( 'switch_theme', array( $this, 'mark_task_as_complete' ) );
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php]: Nitpick: is it possible to bring the "prevent" logic here?

```suggestion
		$theme_switch_via_cys_ai_loader = 1 === absint( $request['theme_switch_via_cys_ai_loader'] );
		if ( ! $theme_switch_via_cys_ai_loader ) {
				add_action( 'switch_theme', array( $this, 'mark_task_as_complete' ) );
		}
```

Comment (chihsuan) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php]: Yes! 👍  Updated in 5a1c3c697c11c22608b3ef6d966e835ad2796fba.

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

**Review by adrianduffell — APPROVED** [review]
It's brilliant @chihsuan, thank you! 🚀

**Pull request merged**